Python/sysmodule.c

```diff
@@ -951,10 +951,6 @@ static int
 profile_trampoline(PyObject *self, PyFrameObject *frame,
                    int what, PyObject *arg)
 {
-    if (arg == NULL) {
```

Comment:
Perhaps turn this check into an `assert` to make clear what the invariant is supposed to be?

Reply:
It could crash interpreter, since `arg` can be set to `NULL`, e. g. in `_PyEval_EvalFrameDefault` during exception unwinding when profille function is set
https://github.com/python/cpython/blob/50a70a083d34305a52fac4f5901bff2ead152d68/Python/ceval.c#L885-L890
https://github.com/python/cpython/blob/50a70a083d34305a52fac4f5901bff2ead152d68/Python/ceval.c#L5127-L5147

My point is that `trace_trampoline` and `profile_trampoline` are local functions which are used only as callbacks for `_PyEval_SetProfile`/`PyEval_SetProfileAllThreads` calls in `sys_settrace`/`sys__settraceallthreads` and `sys_setprofile`/`sys__setprofileallthreads` respectively, and they are both using `call_trampoline`, which does `NULL` check by its own.
Maybe it would be better to even make `call_trampoline` inline or turn it into macro.

Reply:
Thanks for explaining.

I misunderstood and thought that the check is redundant, because the variable can never be null. But it's actually redundant, because we check later.

All makes sense now.

Reply:
Thanks for review. :)

Reply:
I'd leave the conversion to Py_None here, and replace the one in call_trampoline by an assertion.